FAQ/revocation.md

```diff
@@ -0,0 +1,38 @@
+# Revocation FAQ (draft)
+
+Starting from v1.2.0 of the SMART Health Card (SHC) framework, individual health cards issued by mistake can be revoked by listing its revocation identifier in an issuer's revocation list. Legacy health cards can use an external mechanism to derive a revocation identifier based on the health card's content; see the [legacy revocation methods](#legacy-revocation-methods) section.
+
+## Main revocation method
+
+### What should I use as a revocation identifier?
+
+Issuers that keep track of every single issued SHC could create a per-SHC `rid` for fine-grained revocation. In many cases, an issuer will have an internal user ID that can be used to revoke all cards belonging to a particular user; using the timestamp feature allows an issue to invalidate cards up to a certain time.
+
+### Why is a one-way transformation on the user ID recommended for revocation ID?
+
+Publishing an internal user ID might be a privacy issue. A one-way transform with high-entropy input prevents reversal of the CRL’s content. The proposed HMAC-SHA-256 algorithm using a 256-bit key achieves that.
+
+### Why are Card Revocation Lists tied to a key identifier?
+
+Since SHC don’t have expiry dates, public keys and revocation information must be publicly available forever. Creating a per-kid CRL allows issuers to cap the size of CRLs, and verifier apps might not need to download the CRLs of old keys when the corresponding SHCs are replaced by newer ones.
```

Comment:
Can we allow an optional `exp` field to designate an expiration date for an SHC?  This may not be needed in general but it can be very useful to enable a country to issue temporary vaccination certificates to foreign visitors.

Reply:
The `exp` field is already usable.

Per the existing [SHC Modeling spec](https://spec.smarthealth.cards/credential-modeling/#mapping-into-the-w3c-vc-data-model):
> Process the payload according to [JWT Decoding Rules](https://www.w3.org/TR/vc-data-model/#jwt-decoding)

Le linked page specifies `exp` and `expirationDate` that matches that use case.

Reply:
Valid question, but should be discussed separately. You might want to open an issue on the spec to discuss @pvillela. Even if the VC spec supports it, you won't see validation apps processing this field unless the SHC explains it.

Reply:
@christianpaquin, can you send me a link to where I can open the issue?

Reply:
You can open an issue right on this github repository. See also this [zulip](https://chat.fhir.org/#narrow/stream/284830-smart.2Fhealth-cards/topic/SHC.20Expiration.20Claim) thread discussing SHC expiration.